core/utils/histogram.h

```diff
@@ -134,18 +138,65 @@ class Histogram {
     return ret;
   }
 
+  size_t num_buckets() const { return buckets_.size(); }
+  T bucket_width() const { return bucket_width_; }
+
+  size_t max_num_buckets() const {
+    size_t system_limit = buckets_.max_size();
+    // Want std::min(kMaxBuckets, system_limit), but that's by-reference,
+    // which causes link failures at -O0.  See also
+    // https://stackoverflow.com/questions/5391973/
+    return system_limit < kMaxBuckets ? system_limit : kMaxBuckets;
+  }
+
   // Resets all counters, and the count of such counters.
   // Note that the number of buckets remains unchanged.
   void Reset() {
     count_ = 0;
     std::fill(buckets_.begin(), buckets_.end(), T());
   }
 
+  // Resize the histogram.  Note that this resets it (i.e., this
+  // does not attempt to redistribute existing counts).
+  void Resize(size_t num_buckets, T bucket_width) {
+    buckets_.clear();
+    buckets_.resize(num_buckets + 1);
+    bucket_width_ = bucket_width;
+    count_ = 0;
+
+    // new_buckets_ should generally have no owned storage, but in case
+    // someone did a PrepareFastReset() and failed to call FastReset(),
+    // discard storage if we have any.
+    new_buckets_.clear();
+    new_buckets_.shrink_to_fit();
+  }
+
+  // Prepare to do a fast reset.  May be called outside critical
```

Comment:
I just wanted to hide away the `std::move` that is currently in `Measure::Clear()` that requires declaring two local temporary histogram instances. The old way felt klunky. The new way feels klunky. I'm not at all sure there's any improvement 😛

Reply:
Then, if you don't mind, could you revert it back to the old `Measure::Clear()`? I am not sure if the "flipping" operation is intrinsic to the `Histogram` class, just like `std::map` doesn't have thread-safe `insert()`. While it's arguable, I think these should be done by library users, not the library itself.

I do agree that both approaches are clunky. But clunky user must be better than clunky library, so that we can accommodate other non-clunky users later.